### ISSUE TITLE
Calendar: Input element should not have the role attribute set to `combobox` [Accessibility]

### ISSUE THREAD

**Issue by Jogonarr-3Pillar** [primefaces/primeng#15924]
### Describe the bug

The input shouldn't have the role set to `combobox`.

![image](https://github.com/primefaces/primeng/assets/39072072/c15779f3-4c43-407e-bebe-bda72988ab8c)

The role should be removed so screen-readers narrate the element as an "edit" like this example:
![image](https://github.com/primefaces/primeng/assets/39072072/b32b429a-26af-4757-a3cf-4457e203281b)

### Environment

Requires a screen-reader tool installed (tested witn NVDA v2024.1.0.31547) to be able to test how it is being narrated.

Can be tested using the Basic example from the documentation:
[PrimeNg - Calendar - Basic](https://primeng.org/calendar#basic)

### Reproducer

https://stackblitz.com/edit/github-uneuvd?file=src%2Fapp%2Fapp.component.html

### Angular version

17.3.11

### PrimeNG version

17.18.2

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.11.1

### Browser(s)

Chrome 125.0.6422.142

### Steps to reproduce the behavior

1. Open a screen-reader (tested wth NVDA v2024.1.0.31547)
2. Tab into the calendar input
3. Notice that it narrates "Button Choose Month Button Choose Year"

### Expected behavior

Should narrate input as an "edit" instead of a "combobox".

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
